core/app/models/concerns/spree/translatable_resource_slug.rb

```diff
@@ -4,12 +4,10 @@ module TranslatableResourceSlug
 
     included do
       def localized_slugs_for_store(store)
-        localized_slugs = Hash[translations.pluck(:locale, :slug)]
-        default_locale = store.default_locale
         supported_locales = store.supported_locales_list
 
         supported_locales.each_with_object({}) do |locale, hash|
-          hash[locale] = localized_slugs[locale] || localized_slugs[default_locale]
+          hash[locale] = I18n.with_locale(locale) { slug }
```

Comment:
Technically, we could loosen the contract between this module and the including module by introducing a dependency on `FriendlyId`. I think you can access the `slug_column` config value by doing `model.friendly_id_config.slug_column`. Not great either, but it popped in my mind after seeing that you had to add a `slug` alias to the Taxon model just so this works. Something to consider in the future maybe

Reply:
We wanted to unify the naming of the `permalink` -> `slug` column anyway, as it's a bit confusing right now. Technically all of them should be called `slug`, as `permalink` refers to the whole URL, including domain name.